### PR TITLE
Option to return NaNs in compute_correlation

### PR DESCRIPTION
**Pull request by snastase**
I've added a `return_nans` option to `brainiak.fcma.util.compute_correlation`. Original behavior was to return float 0.0 correlation for any vectors containing NaNs. Now, if `return_nans=False` (the default) we the original behavior, but if `return_nans=True` all zeros are replaced with NaNs before output. This is a very simplistic approach: Another way to do this would be to check the input matrices for NaNs and use this to supply NaNs in the output if `return_nans=True`. For example, if you somehow got a "true" correlation of zero, this would replace the zero with NaN even if there wasn't a NaN in the input. This PR references issue #400.

### CODE REVIEW

File: brainiak/fcma/util.py [hunk 1]
```diff
@@ -115,4 +121,9 @@ def compute_correlation(matrix1, matrix2):
                                               0.0,
                                               corr_data,
                                               r2)
+
+    # optionally convert zeros back to NaNs
+    if return_nans:
+        corr_data[corr_data == 0] = np.nan
```
Comment (yidawang) [inline brainiak/fcma/util.py]: It is possible that we have a pair whose correlation is 0. You may want to add this check to line 50, in `_normalize_for_correlation`

Comment (snastase) [reply brainiak/fcma/util.py]: Good point—makes sense. New commit seems to behave as expected.

**Comment by snastase** [comment]
I'm having a hard time understanding the test failures here—seems to be something to do with theano used by test_sssrm...? This is not modified in this PR. Any ideas? @mihaic

**Comment by mihaic** [comment]
See PR #408.

**Pull request merged**